examples/scripts/ct_fan_svmbir_ppp_bm3d_admm_prox.py

```diff
@@ -0,0 +1,285 @@
+"""
+
+
+def add_poisson_noise(sino, max_intensity):
+    expected_counts = max_intensity * np.exp(-sino)
+    noisy_counts = np.random.poisson(expected_counts).astype(np.float32)
+    noisy_counts[noisy_counts == 0] = 1  # deal with 0s
+    y = -np.log(noisy_counts / max_intensity)
+
+    return y
+
+
+y_fan = add_poisson_noise(sino_fan, max_intensity=500)
+
+
+"""
+Reconstruct using default prior of SVMBIR :cite:`svmbir-2020`.
+"""
+weights_fan = svmbir.calc_weights(y_fan, weight_type="transmission")
+
+x_mrf_fan = svmbir.recon(
+    np.array(y_fan[:, np.newaxis]),
+    np.array(angles),
+    weights=weights_fan[:, np.newaxis],
+    num_rows=N,
+    num_cols=N,
+    positivity=True,
+    verbose=0,
+    stop_threshold=0.0,
+    geometry="fan",
+    dist_source_detector=dist_source_detector,
+    magnification=magnification,
+    delta_channel=1.0,
+    delta_pixel=1.0 / magnification,
+)[0]
+
+x_mrf_parallel = svmbir.recon(
+    np.array(y_fan[:, np.newaxis]),
+    np.array(angles),
+    weights=weights_fan[:, np.newaxis],
+    num_rows=N,
+    num_cols=N,
+    positivity=True,
+    verbose=0,
+    stop_threshold=0.0,
+    geometry="parallel",
+)[0]
+
+
+"""
+Push arrays to device.
+"""
+y_fan, x0_fan, weights_fan = jax.device_put([y_fan, x_mrf_fan, weights_fan])
+x0_parallel = jax.device_put(x_mrf_parallel)
+
+
+"""
+Set problem parameters and BM3D pseudo-functional.
+"""
+ρ = 10  # ADMM penalty parameter
+σ = density * 0.6  # denoiser sigma
+g0 = σ * ρ * BM3D()
+
+
+"""
+Set up problem using `SVMBIRExtendedLoss`.
+"""
+f_extloss_fan = SVMBIRExtendedLoss(
+    y=y_fan,
+    A=A_fan,
+    W=Diagonal(weights_fan),
+    scale=0.5,
+    positivity=True,
+    prox_kwargs={"maxiter": 5, "ctol": 0.0},
+)
+f_extloss_parallel = SVMBIRExtendedLoss(
+    y=y_fan,
+    A=A_parallel,
+    W=Diagonal(weights_fan),
+    scale=0.5,
+    positivity=True,
+    prox_kwargs={"maxiter": 5, "ctol": 0.0},
+)
+
+solver_extloss_fan = ADMM(
+    f=None,
+    g_list=[f_extloss_fan, g0],
+    C_list=[Identity(x_mrf_fan.shape), Identity(x_mrf_fan.shape)],
+    rho_list=[ρ, ρ],
+    x0=x0_fan,
+    maxiter=20,
+    subproblem_solver=LinearSubproblemSolver(cg_kwargs={"tol": 1e-3, "maxiter": 100}),
+    itstat_options={"display": True},
+)
+solver_extloss_parallel = ADMM(
+    f=None,
+    g_list=[f_extloss_parallel, g0],
+    C_list=[Identity(x_mrf_parallel.shape), Identity(x_mrf_parallel.shape)],
+    rho_list=[ρ, ρ],
+    x0=x0_parallel,
+    maxiter=20,
+    subproblem_solver=LinearSubproblemSolver(cg_kwargs={"tol": 1e-3, "maxiter": 100}),
+    itstat_options={"display": True},
+)
+
+
+"""
+Run the ADMM solvers.
+"""
+print(f"Solving on {device_info()}\n")
+x_extloss_fan = solver_extloss_fan.solve()
+hist_extloss_fan = solver_extloss_fan.itstat_object.history(transpose=True)
+
+print()
+x_extloss_parallel = solver_extloss_parallel.solve()
+hist_extloss_parallel = solver_extloss_parallel.itstat_object.history(transpose=True)
+
+
+"""
+Show the recovered images.
```

Comment:
Worth adding a note here explaining why the parallel beam reconstructions is so bad. The reason is alluded to in the header comment, but a reminder here wouldn't hurt.

Reply:
Fixed it.

Reply:
I can't readily see the reconstructions. I think this would be most convincing if you pick a small cone angle such that the parallel reconstruction is a bit distorted, but not total nonsense.

Reply:
Is there some issue with the plot display? Works for me, but I run it in as a notebook (due to lack of admin access).

Reply:
I think @Michael-T-McCann's point is that the reconstruction quality is really bad, not that he can't see the plots at all.

Reply:
That is a good point. I have reduced the cone angle so that the parallel beam approximation is better and thus the parallel-beam reconstruction also better.